core/domain/exp_fetchers_test.py

```diff
@@ -1013,3 +1013,23 @@ def test_migration_then_reversion_maintains_valid_exploration(self) -> None:
         self.assertEqual(
             exploration.to_yaml(), '%sversion: 6\n' % self.UPGRADED_EXP_YAML)
         exploration.validate()
+
+    def test_get_user_progress_in_exploration(self) -> None:
+        auth_id: str = 'test_id'
+        username: str = 'testname'
+        user_email: str = 'test@email.com'
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        user_services.set_username(user_id, username)
+        exp_id = exp_fetchers.get_new_exploration_id()
+        self.save_new_valid_exploration(exp_id, user_id)
+
+        user_services.update_learner_checkpoint_progress(
+            user_id, exp_id, 'Introduction', 1)
+
+        user_progress = exp_fetchers.get_user_progress_in_exploration(
+            user_id, [exp_id])
+
+        self.assertEqual(len(user_progress), 1)
+        self.assertEqual(user_progress[0]['exploration_id'], exp_id)
+        self.assertEqual(user_progress[0]['visited_checkpoints_count'], 1)
+        self.assertEqual(user_progress[0]['total_checkpoints_count'], 1)
```

Comment:
Should there be pre-checks?

Reply:
I think no.Please let me know if you want